docs/source/tutorials/improving_ml_performance.ipynb

```diff
@@ -0,0 +1,3573 @@
+    "This tutorial demonstrates a way to use `cleanlab` (via `Datalab`) to curate both your training and test data, ensuring **robust model training** and **reliable performance evaluation**.\n",
+    "\n",
+    "Here's how we recommend handling noisy training and test data (this tutorial walks through these steps):\n",
+    "\n",
+    "1. Preprocess your training and test data. Use cleanlab to check for issues in the merged dataset like train/test leakage or drift.\n",
+    "2. Fit your ML model to your noisy training data and get its predictions/embeddings for your test data. Use these model outputs with cleanlab to detect issues in your **test** data.\n",
+    "3. Manually review/correct cleanlab-detected issues in your test data. To avoid bias, **we caution against automated correction of test data**. Test data changes should be individually verified to ensure they will lead to more accurate model evaluation. We also caution against comparing the performance of different ML models across different versions of your test data; performance comparions between models should be based on the same test data.\n",
+    "4. Cross-validate a new copy of your ML model on your training data, and then use it with cleanlab to detect issues in the **training** dataset. Do not include test data in any part of this step to avoid leaking test set information into the training data curation.\n",
+    "5. You can try **automated techniques** to curate your training data based on cleanlab results, train models on the curated training data, and evaluate them on the cleaned test data.\n",
+    "\n",
+    "Consider this tutorial as a blueprint for using cleanlab in diverse ML projects spanning various data modalities. In a final advanced section of this tutorial, we show how training data edits can be parameterized in terms of cleanlab's detected issues, such that hyperparameter optimization can identify the optimal combination of data edits for training an effective ML model.\n",
+    "\n",
+    "**Note**: This tutorial uses tabular data, but the same approach applies to *any* data modality for which you can train a ML model. While we use an XGBoost model here, you can apply these steps with *any* ML model."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "72907260",
+   "metadata": {},
+   "source": [
+    "## 1. Install required dependencies\n",
+    "\n",
+    "`Datalab` has additional dependencies that are not included in the standard installation of cleanlab.\n",
+    "You can use `pip` to install all packages required for this tutorial as follows:\n",
+    "\n",
+    "```ipython3\n",
+    "!pip install xgboost\n",
+    "!pip install \"cleanlab[datalab]\"\n",
+    "# Make sure to install the version corresponding to this tutorial\n",
+    "# E.g. if viewing master branch documentation:\n",
+    "#     !pip install git+https://github.com/cleanlab/cleanlab.git\n",
```

Comment:
ah this is pretty complex. Look at our other tutorials on docs.cleanlab.ai, in Colab, and in local jupyter notebook.

You'll notice 3 versions of this: it's hidden on our docs website, in colab it's auto-filled with the right version to install (based on whether you are viewing master-branch docs or stable release docs). 

The complexity is necessary because we don't know where you came to this doc from (stable release, old cleanlab version, latest master version).  Open to other suggestions for the verbiage here, but be careful if you change any of the actual code (even the commented part), because it's being auto-replaced in CI to make this work for across  cleanlab versions.

Reply:
also recommend you open the raw .ipynb file in a text editor (or in GH view raw file), and look at the metadata of each notebook cell. You'll notice some of them say hidden which means theyre hidden on docs.cleanlab.ai